src/acrcssc/azext_acrcssc/helper/_taskoperations.py

```diff
@@ -0,0 +1,568 @@
+            registry.id,
+            filter=f"principalId eq '{identity.principal_id}'"
+        )
+
+        for role in assigned_roles:
+            if dryrun:
+                logger.debug(f"Dry run, skipping deletion of role assignments, task: {task_name}, role name: {role.name}")
+                return None
+            logger.debug(f"Deleting role assignments of task {task_name} from the registry")
+            role_client.role_assignments.delete(
+                scope=registry.id,
+                role_assignment_name=role.name
+            )
+
+
+def _transform_task_list(tasks):
+    transformed = []
+    for task in tasks:
+        if task.name not in CONTINUOUSPATCH_ALL_TASK_NAMES:
+            continue
+
+        transformed_obj = {
+            "creationDate": task.creation_date,
+            "location": task.location,
+            "name": task.name,
+            "provisioningState": task.provisioning_state,
+            "systemData": task.system_data,
+            "schedule": None,
+            "description": CONTINUOUSPATCH_TASK_DEFINITION[task.name][DESCRIPTION]
+        }
+
+        trigger = task.trigger
+        if trigger and trigger.timer_triggers:
+            # convert the cron expression to a 'days' format to keep consistent with the command's options
+            transformed_obj["schedule"] = convert_cron_to_schedule(trigger.timer_triggers[0].schedule)
+
+            # add a 'nextOccurrence' field to the task, only for the scheduling task
+            transformed_obj["nextOccurrence"] = get_next_date(task.trigger.timer_triggers[0].schedule)
+        transformed.append(transformed_obj)
+
+    return transformed
+
+
+def _get_custom_registry_credentials(cmd,
+                                     auth_mode=None,
+                                     login_server=None,
+                                     username=None,
+                                     password=None,
+                                     identity=None,
+                                     is_remove=False):
+    """Get the credential object from the input
+    :param str auth_mode: The login mode for the source registry
+    :param str login_server: The login server of custom registry
+    :param str username: The username for custom registry (plain text or a key vault secret URI)
+    :param str password: The password for custom registry (plain text or a key vault secret URI)
+    :param str identity: The task managed identity used for the credential
+    """
+    acr_tasks_client = cf_acr_tasks(cmd.cli_ctx)
+    source_registry_credentials = None
+    if auth_mode:
+        source_registry_credentials = acr_tasks_client.models.SourceRegistryCredentials(
+            login_mode=auth_mode)
+
+    custom_registries = None
+    if login_server:
+        # if null username and password (or identity), then remove the credential
+        custom_reg_credential = None
+
+        is_identity_credential = False
+        if not username and not password:
```

Comment:
Question: what happen if username is none, but password is not none?

Reply:
I think this is picked up from the ACR module in the base CLI, and not all parameters are being used
https://github.com/Azure/azure-cli/blob/6f3fa67f6544566b564bef5d61fcc0a7ddc24550/src/azure-cli/azure/cli/command_modules/acr/_utils.py#L281

will remove parameters and code not being used to generate `CustomRegistryCredentials`